c/tskit/tables.c

```diff
@@ -6982,7 +6982,7 @@ simplifier_merge_ancestors(simplifier_t *self, tsk_id_t input_id)
         keep_unary = true;
     }
     if ((self->options & TSK_KEEP_UNARY_IN_INDIVIDUALS)
-        && (self->tables->nodes.individual[input_id] != TSK_NULL)) {
+        && (self->input_tables.nodes.individual[input_id] != TSK_NULL)) {
```

Comment:
This was a C bug, so as a minimum we need to add a C test that provoked the original bug

Reply:
Hmm, I think that might not be so easy. Part of the reason for exposing this to python was that it is much easier to test for this bug (the python tests I introduced fail unless the fix it there).

For the C tests, I suppose one way would be change the hard-coded example in https://github.com/tskit-dev/tskit/blob/028997baf34de806674309c9dc84297bc71b2d02/c/tests/test_trees.c#L1054 so it triggers this bug. It might take a bit of fiddling to find an example that does this, though (and the bug is dependent on the node visiting order within simplify(), I think)

Reply:
As discussed, the code without the C fix reliably fails the python tests, and the old C code didn't segfault, so I think we're covered.